app/assets/javascripts/host_edit.js

```diff
@@ -2,6 +2,43 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
 $(document).on('AddedClass', function(event, link){load_puppet_class_parameters(link)});
 $(document).on('click', '#params-tab', function() { resizeTextareas($('#params')); });
 
+LookupValueController = function(item) {
+  var self = this; //because JS' this confuses me
```

Comment:
for instance, the this variable gets replaced in the callbacks, so I don't see how I can put the callbacks in prototype. I may not know JS well enough though...